action/lib/Database.js

```diff
@@ -1,206 +1,60 @@
 const common = require('./common');
 
-// constructor for DB object - a thin, promise-loving wrapper around nano
-module.exports = function(dbURL, dbName) {
-    var nano = require('nano')(dbURL);
-    this.db = nano.db.use(dbName);
-    var utilsDB = this;
+module.exports = function() {
 
-    this.getWorkerID = function(availabeWorkers) {
-
-        return new Promise((resolve, reject) => {
-            var workerID = availabeWorkers[0] || 'worker0';
-
-            if (availabeWorkers.length > 1) {
-                utilsDB.db.view('triggerViews', 'triggers_by_worker', {reduce: true, group: true}, function (err, body) {
-                    if (!err) {
-                        var triggersByWorker = {};
+    var database = this;
 
-                        availabeWorkers.forEach(worker => {
-                            triggersByWorker[worker] = 0;
-                        });
+    // init for DB object - a thin, promise-loving wrapper around nano / documentdb
+    this.initDB = function(dbURL, dbName, dbType, cosmosdbRootDatabase, cosmosdbMasterKey) {
 
-                        body.rows.forEach(row => {
-                            if (row.key in triggersByWorker) {
-                                triggersByWorker[row.key] = row.value;
-                            }
-                        });
-
-                        // find which worker has the least number of assigned triggers
-                        for (var worker in triggersByWorker) {
-                            if (triggersByWorker[worker] < triggersByWorker[workerID]) {
-                                workerID = worker;
-                            }
-                        }
-                        resolve(workerID);
-                    } else {
-                        reject(err);
-                    }
-                });
+        dbType = typeof dbType  !== 'undefined' ?  dbType  : "couchdb";
+        var db = {};
+        return new Promise((resolve, reject) => {
+            if(dbType === "couchdb") {
+                console.log("using couchdb");
+                var couchdb = require('./couchdb');
+                db = new couchdb(dbURL, dbName);
+                database.utilsDB = db;
+                resolve();
             }
-            else {
-                resolve(workerID);
+            else if(dbType === "cosmosdb") {
+                console.log("using cosmosdb");
+                var cosmosdb = require('./cosmosdb');
+                db = new cosmosdb(dbURL, cosmosdbMasterKey);
```

Comment:
instead of duplicating functions like `createTrigger`, I think we should duplicate document manipulation functions like `createDocument` so that the db operations are the only piece that is delegated to the db impls.